### PR TITLE
Add alignment via STAR + postprocessing

### PR DESCRIPTION
**Pull request by pinin4fjords**
PR to add STAR to produce both genome and transcriptome-level alignment to the workflow, using the fastq_align_star subworkflow which also gives us indexing and stats generation.

Also pulled the UMI deduplication from nf-core rna-seq, though testing will have to wait until I identify some UMI-tagged riboseq data to play with.

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/riboseq/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/riboseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: conf/modules.config [hunk 1]
```diff
@@ -534,8 +534,19 @@ if (!params.skip_alignment) {
...
+//                '--alignMatesGapMax 5000',
+//                '--outFilterMismatchNmax 2',
+//                '--outFilterScoreMin 20',
+//                '--outFilterMatchNmin 16',
+//                '--seedSearchStartLmax 10',
+//                '--alignSJDBoverhangMin 1',
+//                '--alignSJoverhangMin 8',
+//                '--clip3pNbases 0',
+//                '--clip5pNbases 0',
+//                '--outSAMunmapped Within',
```
Comment (maxulysse) [inline conf/modules.config:549]: why commented?

File: conf/test_full.config [hunk 2]
```diff
@@ -1,11 +1,11 @@
 /*
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    Nextflow config file for running full-size tests
+    Nextflow config file for running minimal tests
```
Comment (maxulysse) [inline conf/test_full.config:3]: this is test full

File: subworkflows/local/deduplicate_bam_by_umi/main.nf [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    UMITOOLS_PREPAREFORSALMON (
+        ch_umitools_dedup_bam.paired_end
+    )
+    ch_versions = ch_versions.mix(UMITOOLS_PREPAREFORSALMON.out.versions.first())
+
+    ch_umitools_dedup_bam
+        .single_end
+        .mix(UMITOOLS_PREPAREFORSALMON.out.bam)
+        .set { ch_transcriptome_bam }
+}
```
Comment (maxulysse) [inline subworkflows/local/deduplicate_bam_by_umi/main.nf:64]: no emit statement?

File: subworkflows/nf-core/preprocess_rnaseq/.main.nf.swp
Comment (maxulysse) [inline subworkflows/nf-core/preprocess_rnaseq/.main.nf.swp:1]: what is this file?

File: conf/modules.config [hunk 4]
```diff
@@ -534,16 +534,17 @@ if (!params.skip_alignment) {
...
                 '--quantMode TranscriptomeSAM',
                 '--twopassMode Basic',
                 '--outSAMtype BAM Unsorted',
                 '--readFilesCommand zcat',
                 '--runRNGseed 0',
                 '--outFilterMultimapNmax 20',
                 '--alignSJDBoverhangMin 1',
-                '--outSAMattributes NH HI AS NM MD',
                 '--quantTranscriptomeBan Singleend',
                 '--outSAMstrandField intronMotif',
```
Comment (maxulysse) [inline conf/modules.config:549]: is the order of any importance?

File: conf/test_full.config (hunk 2 above)
Comment (pinin4fjords) [reply conf/test_full.config]: Thanks!

File: subworkflows/local/deduplicate_bam_by_umi/main.nf (hunk 3 above)
Comment (pinin4fjords) [reply subworkflows/local/deduplicate_bam_by_umi/main.nf]: This is a relic- didn't end up using it and forgot to remove, thanks!

File: subworkflows/nf-core/preprocess_rnaseq/.main.nf.swp
Comment (pinin4fjords) [reply subworkflows/nf-core/preprocess_rnaseq/.main.nf.swp:1]: Vim swap file- need to go, thanks!

File: conf/modules.config (hunk 4 above)
Comment (pinin4fjords) [reply conf/modules.config:549]: Of the CLI args? Don't think so

Comment (maxulysse) [reply conf/modules.config:549]: Then why don't we order it?
(You know I like a logical order, alphabetical or any other one as long as it makes sense)

File: conf/test_full.config [hunk 5]
```diff
@@ -2,10 +2,10 @@
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     Nextflow config file for running full-size tests
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-    Defines input files and everything required to run a full size pipeline test.
+    Defines input files and everything required to run a fast and simple pipeline test.
 
     Use as follows:
-        nextflow run nf-core/riboseq -profile test_full,<docker/singularity> --outdir <OUTDIR>
+        nextflow run nf-core/riboseq -profile test,<docker/singularity> --outdir <OUTDIR>
```
Comment (maxulysse) [inline conf/test_full.config:8]:
```suggestion
        nextflow run nf-core/riboseq -profile test_full,<docker/singularity> --outdir <OUTDIR>
```

File: conf/modules.config (hunk 4 above)
Comment (pinin4fjords) [reply conf/modules.config]: Oh, alright. [Happy now](https://github.com/nf-core/riboseq/pull/12/commits/d7198434940ee2a23ad92cb4f3d6aa6911eddc3e)? :-D

**Review by maxulysse — APPROVED** [review]
LGTM

**Pull request merged**